Theme refactoring and csv upload

marillam opened the pull request:
Conflicts resolved now, so should be ready for review. Please check that the frontend runs as you would expect in this branch! (Attainment and formula sections are not so familiar to me)

This pr contains:
- theme refactoring (mainly defining the theme to match our guidelines and then changes to headings)
- a MVP for civ upload / the button to select way to import grades
- the course view page also had some small UI changes (new splitting for the heading so long courses look better)
- changes to test file so the csv upload is included

viivi-uhari reviewed (changes requested):
Overall great work! However, the implementation should be looked over to better match the design. 

Also, a couple of other notes here because I couldn't mention them by commenting the changes:

Found a couple of fontSizes still in other files than App.js. For example, TextFieldBox.js and SelectFormulaForm.js still specify font sizes for input labels as numbers/values instead of using the theme so could/should these be looked over?

The phrase `component="div"` can also be found from FormulaAttributesView.js. This can also be found from the List component in ParentAssignment.js (line 83) and it can be deleted from there as well.

Finally, there are some comment with the word 'assignment' instead of 'attainment' that could be changed if you're up for it.
App.js (line :
`{ /* '/:assignmentId' will be added to the paths above once they work for sub-assignments */ }`
ParentAssignment.js (line 35): 'assignments' -> 'attainment'
CourseView.js (line 95): 'assignment' -> 'attainment'
CourseView.test.js (line 76): 'assignments' -> 'attainments'

kiiamakir reviewed (changes requested):
Looks good! I agree with @viivi-uhari about her suggestions about further changes.

Also SelectFormulaForm still uses color: #000 on line 162, this should be changed to use the color defined in the theme.

kiiamakir commented on client/src/components/course-view/FileLoadDialog.js:
```diff
@@ -0,0 +1,69 @@
+
+// A Dialog component for uploading a file
+
+const instructions = 'Upload a CSV file with the header "studentNo" and headers matching to the study \
```
I agree that the grey needs to be changed to fit accessibility standards. I feel like the black that MUI uses is too close to black and thus it won't be noticable. Could we pick something that's in between #707070 and the MUI black? For example #545454

marillam replied:
Picture added and went with #545454. I added it to the theme as infoGrey.

marillam commented on client/src/components/course-view/FileLoadDialog.js:
```diff
@@ -0,0 +1,69 @@
+        <DialogContentText sx={{ mb: 3 }}>{instructions}</DialogContentText>
+        <Box sx={{ display: 'flex', flexWrap: 'wrap', justifyContent: 'flex-start', alignItems: 'center', columnGap: 2 }}>
+          <Button variant='contained' component='label'>
+            Upload file
```
I think we might want to consider keeping this as contained. I think it highlights that this is the primary action button on this dialog. (I did change it now to match the designs)

marillam commented on client/src/components/course-view/MenuButton.js:
```diff
@@ -0,0 +1,114 @@
+    }}
+    {...props}
+  />
+))(({ theme }) => ({
```
Maybe, but currently all of our styled components have been defined like this. I think the inspiration is from a mui example so I'll keep this as is for now.

marillam commented:
> Looks good! I agree with @viivi-uhari about her suggestions about further changes.
> 
> Also SelectFormulaForm still uses color: #000 on line 162, this should be changed to use the color defined in the theme.

- done

marillam commented:
> Overall great work! However, the implementation should be looked over to better match the design.
> 
> Also, a couple of other notes here because I couldn't mention them by commenting the changes:
> 
> Found a couple of fontSizes still in other files than App.js. For example, TextFieldBox.js and SelectFormulaForm.js still specify font sizes for input labels as numbers/values instead of using the theme so could/should these be looked over?
> 
> The phrase `component="div"` can also be found from FormulaAttributesView.js. This can also be found from the List component in ParentAssignment.js (line 83) and it can be deleted from there as well.
> 
> Finally, there are some comments with the word 'assignment' instead of 'attainment' that could be changed if you're up for it. App.js (line 173): `{ /* '/:assignmentId' will be added to the paths above once they work for sub-assignments */ }` ParentAssignment.js (line 35): 'assignments' -> 'attainment' CourseView.js (line 95): 'assignment' -> 'attainment' CourseView.test.js (line 76): 'assignments' -> 'attainments'

fontSIzes: 
- SelectFormulaField changed (fyi the syntax is ''h3.fontSize" since the component isn't typography.
- TextFieldBox was more difficult and we don't have 32px fontSize defined in our theme. I decided to use the h2 (34px) as this was a good way to demo how to access the theme in this different situation ("h2.fontSize" didn't work). The point: use useTheme hook if theme doesn't work otherwise.
- The others I found were controlling icon sizes I think

component="div"
- deleted from the mentioned files

assignment: changed to attainment. While I was doing this I realised that quite a few of our files and components are still named as assignments. But I think this should be changed in another branch if we decide to refactor

viivi-uhari commented:
> assignment: changed to attainment. While I was doing this I realised that quite a few of our files and components are still named as assignments. But I think this should be changed in another branch if we decide to refactor

Yeah the idea was that we first change the variables and functions, and that we change the components and folders in another branch only dedicated for this.

viivi-uhari commented on client/src/components/course-view/FileLoadDialog.js:
```diff
@@ -0,0 +1,104 @@
+  return (
+    <Dialog open={open} >
+      <DialogTitle >Add Grades from File</DialogTitle>
+      <DialogContent sx={{ pr: 6 }}>
```
The padding here is a bit odd. Because of `pr` the content isn't in the middle, so I would delete the `pr: 6`. I would also delete the bottom padding because there seems to be quite a big of a gap, so I would add `pb: 0`.

marillam replied:
good point!

viivi-uhari commented on client/src/components/course-view/FileLoadDialog.js:
```diff
@@ -0,0 +1,104 @@
+              width: 283,
+              boxShadow: '0px 3px 4px rgba(0, 0, 0, 0.25)'
+            }}
+            alt="A picture of a sheet with three columns. First one is student numbers (StudentNo), the next is points from attainment C3I9A1 and the third column is points from attainment C3I9A2."
```
Great job here with the alt text!

viivi-uhari reviewed (changes requested):
Looks good! just a small change could still be done.
